> Look like the unit tests are broken due to:
> 
> ```python
>         except Exception as ex:
>             ex.add_note(f"Seed: {self.multiworld.seed}")
>             raise
> ```
> 
> in base tests

That's not the root cause. It seems like older versions (Python <= 3.10) don't have the add_note method, but it generally doesn't matter as that just adds information when something has already gone wrong. Actual error on 3.12:

> FAILED worlds/sc2/test/test_usecases.py::TestSupportedUseCases::test_excluding_faction_on_vanilla_order_excludes_epilogue - ValueError: Mission slot at address "Heart of the Swarm/Umoja/0" tried to plando an already plando'd mission.
> Exception in <bound method SC2World.create_regions of <worlds.sc2.SC2World object at 0x10d4f1cd0>> for player 1, named Tester.
> Seed: 52903914536955041501

--- a/worlds/sc2/mission_order/mission_pools.py
+++ b/worlds/sc2/mission_order/mission_pools.py
@@ -105,11 +105,12 @@ class SC2MOGenMissionPools:
     def pull_specific_mission(self, mission: SC2Mission) -> None:
         """Marks the given mission as present in the mission order."""
         # Remove the mission from the master list and whichever difficulty pool it is in
-        self.master_list.remove(mission.id)
-        for diff in self.difficulty_pools:
-            if mission.id in self.difficulty_pools[diff]:
-                self.difficulty_pools[diff].remove(mission.id)
-                break
+        if mission.id in self.master_list:
+            self.master_list.remove(mission.id)
+            for diff in self.difficulty_pools:
+                if mission.id in self.difficulty_pools[diff]:
+                    self.difficulty_pools[diff].remove(mission.id)
+                    break
         self._add_mission_stats(mission)
     
     def _add_mission_stats(self, mission: SC2Mission) -> None:

--- a/worlds/sc2/mission_order/structs.py
+++ b/worlds/sc2/mission_order/structs.py
@@ -405,10 +405,9 @@ class SC2MissionOrder(MissionOrderNode):
             # Remove set mission from locked missions
             locked_ids = [locked for locked in locked_ids if locked != mission_id]
             mission = lookup_id_to_mission[mission_id]
-            try:
-                self.mission_pools.pull_specific_mission(mission)
-            except KeyError:
+            if mission in self.mission_pools.get_used_missions():
                 raise ValueError(f"Mission slot at address \"{mission_slot.get_address_to_mission()}\" tried to plando an already plando'd mission.")
+            self.mission_pools.pull_specific_mission(mission)
             mission_slot.set_mission(world, mission, locations_per_region, location_cache)
             regions.append(mission_slot.region)
 

